File: solutions/cpp/4673/main.cpp
#include<bits/stdc++.h>

using namespace std;

int d(int n) {
    int sum = n;
    while (n != 0) sum += n % 10, n /= 10;
    return sum > 10000 ? 0 : sum;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    vector<int> chk(10001);
    for (int i = 1; i <= 10000; ++i) chk[d(i)] = 1;
    for (int i = 1; i <= 10000; ++i) {
        if (chk[i] != 1) cout << i << '\n';
    }

    return 0;
}

File: solutions/python/4673/main.py
def d(n):
    Sum = n
    while n != 0:
        Sum += n % 10
        n //= 10
    if Sum > 10000:
        Sum = 0
    return Sum


N = 10000
chk = [0] * (N + 1)
for i in range(N + 1):
    chk[d(i)] = 1

print("\n".join([f"{idx}" for idx, x in enumerate(chk) if x != 1]))
